Firebase/InAppMessaging/Analytics/FIRIAMClearcutUploader.m

```diff
@@ -205,15 +202,6 @@ - (void)attemptUploading {
 }
 
 - (void)scheduleNextSend {
-  @synchronized(self) {
-    if (_nextSendScheduled) {
```

Comment:
Don't we want to keep this, checking if `_nextSendScheduled` is already set? I noticed in the log you mentioned it could cause an issue, but is the issue because of this check AND the `scheduleNextSend` in the `init` or just the one?

Reply:
I added this check back, but I'm not sure what `_nextSendScheduled = YES` in the else block was doing.. it's already being set to YES at the end of the method. I removed the else block.

Reply:
I think this could still cause problems if you want to waitTime. The new waitTime will get ignored for the next scheduled send. Perhaps that's ok?

Reply:
I think that should be fine. The timing of sending logs doesn't need to be precise in this case.